numpy/core/src/multiarray/iterators.c

```diff
@@ -1242,232 +1242,183 @@ PyArray_Broadcast(PyArrayMultiIterObject *mit)
     return 0;
 }
 
-/*NUMPY_API
- * Get MultiIterator from array of Python objects and any additional
- *
- * PyObject **mps -- array of PyObjects
- * int n - number of PyObjects in the array
- * int nadd - number of additional arrays to include in the iterator.
- *
- * Returns a multi-iterator object.
+static NPY_INLINE PyObject*
+multiiter_wrong_number_of_args(void)
+{
+    return PyErr_Format(PyExc_ValueError,
+                        "Need at least 1 and at most %d "
+                        "array objects.", NPY_MAXARGS);
+}
+
+/*
+ * Common implementation for all PyArrayMultiIterObject constructors.
+ * This function takes a pointer to an array of at most NPY_MAXARGS
+ * PyObject pointers, NULL terminated unless completely full, which
```

Comment:
Is there any point null terminating here? I'd be inclined to just pass the length separately, and change line 1272 to `for (i = 0; i < n_args; i++)`. Note this loop does not care about `MAX_ARGS`, that's already covered by the `multi->numiter < NPY_MAXARGS` within the loop.